### PR TITLE
Handle Linux Hardening where sudo on root is blocked

### PR DESCRIPTION
**Pull request by simathih**

### CODE REVIEW

File: LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py [hunk 1]
```diff
@@ -94,6 +94,8 @@ def stop_metrics_service(is_lad):
 
         if os.path.isfile(metrics_service_path):
             code = os.system("sudo systemctl stop {0}".format(metrics_service_name))
+            if code != 0:
+                code = os.system("systemctl stop {0}".format(metrics_service_name))
```
Comment (hestolz) [inline LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:98]: do we even need sudo? I do not see it used in `agent.py`

File: LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py [hunk 2]
```diff
@@ -352,8 +356,10 @@ def start_metrics(is_lad):
     if metrics_utils.is_systemd():
         service_restart_status = os.system("sudo systemctl restart {0}".format(metrics_service_name))
         if service_restart_status != 0:
-            log_messages += "Unable to start {0}. Failed to start ME service.".format(metrics_service_name)
-            return False, log_messages
+            service_restart_status = os.system("systemctl restart {0}".format(metrics_service_name))
+            if service_restart_status != 0:
+                log_messages += "Unable to start {0}. Failed to start ME service.".format(metrics_service_name)
```
Comment (NarineM) [inline LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:361]: Looks good, can we append the error message to say "Unable to start ......" + ". Check if the system has some custom hardening or policies preventing systemctl command"

File: LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py (hunk 1 above)
Comment (simathih) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:98]: this code is shared with LAD, I need to check if LAD needs sudo

Comment (simathih) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:98]: Nikhil added the sudo, should we just remove it and see if it breaks anything? :) I checked LAD code and that should be running
 as root as well so sudo shouldn't be needed

Comment (NarineM) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:98]: They can have the opposite when the user is not the root, in this case it should run with sudo.

Comment (simathih) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:98]: user is always root, Guest agent invokes our extension code with root

Comment (simathih) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py:98]: let me remove sudo and I feel we should be fine

Comment (NarineM) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py]: waagent may be, but we are using shim to apply manual workarounds. Is this code also called from shim?

Comment (simathih) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py]: manual workaround can do sudo shim.sh instead...that is for internal use only

Comment (NarineM) [reply LAD-AMA-Common/metrics_ext_utils/metrics_ext_handler.py]: ok, but we will need to check with CSS tsg and update the steps . I am almost certain we have shim in troubleshooting.

**Review by NarineM — APPROVED** [review]

**Pull request merged**